### PR TITLE
UI: HDS adoption replace <Modal> 

### PR DESCRIPTION
**Pull request by hellobontempo**
Side branch for adopting `<Hds::Modal>` 
- This work also replaces all `<button>` components inside modals with `<Hds::Button>` 
- `<Hds::Button>` is now used in `<DownloadButton>`
- Additional `<button>` components are replaced to make way for fully adopting `<Hds::Button>` 
- Adds `@container` to `<Hds::Copy::Button>` components that render inside modals (which are otherwise not functional)

https://github.com/hashicorp/vault/pull/23363
https://github.com/hashicorp/vault/pull/23398
https://github.com/hashicorp/vault/pull/23415
https://github.com/hashicorp/vault/pull/23451
https://github.com/hashicorp/vault/pull/23471
https://github.com/hashicorp/vault/pull/23484
https://github.com/hashicorp/vault/pull/23522

### CODE REVIEW

File: ui/app/styles/core/buttons.scss [hunk 1]
```diff
@@ -332,11 +326,11 @@ a.button.disabled {
   }
 }
 
+// TODO HDS adoption cleanup: audit styles with design and see what to keep/remove once buttons are fully HDS
 // Existing class on <Hds::Copy::Button> component, modifying to match existing UI Structure buttons
 .hds-copy-button {
-  font-weight: $font-weight-semibold;
-  box-shadow: $box-shadow-low;
-  border-radius: $radius;
+  font-weight: $font-weight-semibold; // TODO delete
```
Comment (kiannaquach) [inline ui/app/styles/core/buttons.scss:332]: just out of curiosity - what's the default font-weight for the copy button?

File: ui/package.json [hunk 2]
```diff
@@ -166,7 +165,6 @@
     "ember-test-selectors": "6.0.0",
     "ember-tether": "^2.0.1",
     "ember-truth-helpers": "3.0.0",
-    "ember-wormhole": "0.6.0",
```
Comment (kiannaquach) [inline ui/package.json:169]: 🎉

File: ui/app/styles/core/buttons.scss (hunk 1 above)
Comment (hellobontempo) [reply ui/app/styles/core/buttons.scss:332]: There's no override option, it's only ever the clipboard icon 😄  [[docs](https://helios.hashicorp.design/components/copy/button?tab=code#icon-only)]
Screenshot 2023-10-06 at 1 59 05 PM

**Review by zofskeez — APPROVED** [review]

File: ui/app/styles/core/buttons.scss (hunk 1 above)
Comment (hellobontempo) [reply ui/app/styles/core/buttons.scss:332]: If you consult the docs - the font weights for buttons appear to be `-token-typography-font-weight-regular: 400;`

**Pull request merged**